go/vt/client/fakeserver_test.go

```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/vtgate/proto"
+	"github.com/youtube/vitess/go/vt/vtgate/vtgateservice"
+	"golang.org/x/net/context"
+)
+
+// fakeVTGateService has the server side of this fake
+type fakeVTGateService struct {
```

Comment:
We should talk about it. As I mentioned in the other issue, there are pros & cons.